### ISSUE TITLE
provider/consul: `consul_key_prefix.subkeys` improperly handles keys containing '.'

### ISSUE THREAD

**Issue by gechr** [hashicorp/terraform#6471]
### Terraform Version
`Terraform v0.6.15`

### Affected Resource(s)
- consul_key_prefix

### Terraform Configuration Files
```hcl
resource "consul_key_prefix" "bug" {
  datacenter  = "eu"
  token       = "dummy-token-abcdef"
  path_prefix = "test/prefix/"

  subkeys = {
    "this/is.a.test" = "foobar"
    "this/is/a/test" = "foobar"
    "this-is-a/test" = "foobar"
    "this-is-a-test" = "foobar"
  }
}
```

### Expected Behavior
```
+ consul_key_prefix.bug
    datacenter:             "" => "eu"
    path_prefix:            "" => "test/prefix/"
    subkeys.#:              "" => "4"
    subkeys.this-is-a-test: "" => "foobar"
    subkeys.this-is-a/test: "" => "foobar"
    subkeys.this/is.a.test: "" => "foobar"
    subkeys.this/is/a/test: "" => "foobar"
    token:                  "" => "dummy-token-abcdef"
```

### Actual Behavior
```
+ consul_key_prefix.bug
    datacenter:             "" => "eu"
    path_prefix:            "" => "test/prefix/"
    subkeys.#:              "" => "4"
    subkeys.this-is-a-test: "" => "foobar"
    subkeys.this-is-a/test: "" => "foobar"
    subkeys.this/is.a.test: "" => ""
    subkeys.this/is/a/test: "" => "foobar"
    token:                  "" => "dummy-token-abcdef"
```

### Steps to Reproduce

1. Specify resource `consul_key_prefix` with `subkeys` key containing a `.` character
2. Run `terraform plan` or `terraform apply`

**Comment by gechr** [comment]
Ping @apparentlymart

**Comment by radeksimko** [comment]
This is cause by a known core bug: https://github.com/hashicorp/terraform/issues/2143

**Comment by apparentlymart** [comment]
Yah, unfortunately this (maps with keys that have periods) is just something Terraform can't do for any provider right now...  If the core were fixed I expect this would just stay working.

In the mean time all I can suggest is to choose Consul key names that don't contain periods. Sorry :(

**Comment by radeksimko** [comment]
The way we worked around this in some API Gateway resources recently was to add a field `field_in_json`, which was effectively (un)marshalled under the hood as map (I know, 🐽  yucky, isn't it).

I think that such workarounds should be limited to cases where map keys are not fully in user's control (i.e. keys are prescribed by the API) and also such workarounds will be deprecated once we get the core bug fixed.

**Comment by stack72** [comment]
Hi @gechr 

I believe this is fixed via #2143 :) This will be part of the upcoming 0.7 release so please do let us know if this isn't the case :)

Paul
